[DOCS] Full Disk Access updates [8.0] (#1501)
elastic/security-docs: docs/getting-started/endgame-sensor-FDA.asciidoc

[[endgame-sensor-full-disk-access]]
= Enable Full Disk Access for the Endgame sensor

> Elastic Endgame requires Full Disk Access to protect you from malware and other cybersecurity threats. Full Disk Access permissions is a new privacy feature introduced in macOS Mojave (10.14) that prevents some applications from accessing your data. This means you need to manually grant permission for the Endgame sensor to access these protected areas of your Mac.

Reviewer: ```suggestion
Elastic Endgame requires Full Disk Access to protect you from malware and other cybersecurity threats. Full Disk Access permissions is a new privacy feature introduced in macOS Mojave (10.14) to control which applications can access your data. This means you need to manually grant Full Disk Access permission to the Endgame sensor.
```

[discrete]
[[macos-permissions]]
== macOS permissions

The behavior of the Endgame sensor and Elastic Endpoint differs based on your macOS version. MDM/JAMF users can pre-approve all Full Disk Access without granting permission to the sensor. However, depending on the macOS version and sensor type, non-MDM/JAMF users may be prompted to enable Full Disk Access for required security files.

> Here are the following Full Disk Access requirements for macOS:

Reviewer 2: Maybe clarify that the versions below are macOS versions, not Stack versions:
```suggestion
Here are the following Full Disk Access requirements for specific versions of macOS:
```

> Here are the following Full Disk Access requirements for macOS:

Reviewer: ```suggestion
Here are Elastic's Full Disk Access requirements for different versions of macOS:
```

> - `10.13, 10.14, 10.15`: Users cannot proceed with installation without first granting the sensor the ability to load a <<kernel-ext-approval-endgame, kernel extension>>. During installation, you'll be prompted to go to **System Preferences** and approve loading the kernel. Upon approval, installation proceeds.

Reviewer: ```suggestion
- `10.13, 10.14, 10.15`: Users cannot install the Endgame sensor without allowing it to load a <<kernel-ext-approval-endgame, kernel extension>>. During installation, you'll be prompted to go to **System Preferences** and approve it. Upon approval, installation proceeds.
```

> - `10.13, 10.14, 10.15`: Users cannot proceed with installation without first granting the sensor the ability to load a <<kernel-ext-approval-endgame, kernel extension>>. During installation, you'll be prompted to go to **System Preferences** and approve loading the kernel. Upon approval, installation proceeds.
> 
> - `11.0`, `12.0`: Users cannot proceed with installation without first granting the sensor the ability to load a <<system-extension, system extension>>. During installation, you will be prompted to go to **System Preferences** and approve loading the system extension. Upon approval, a second prompt appears to enable **Network Filtering**. Approve this final prompt for installation to proceed.

Reviewer 2: ```suggestion
- `10.13, 10.14, 10.15`: You cannot install without first granting the sensor the ability to load a <<kernel-ext-approval-endgame, kernel extension>>. During installation, you'll be prompted to go to **System Preferences** and approve loading the kernel. Upon approval, installation proceeds.

- `11.0`, `12.0`: You cannot install without first granting the sensor the ability to load a <<system-extension, system extension>>. During installation, you will be prompted to go to **System Preferences** and approve loading the system extension. Upon approval, a second prompt appears to enable **Network Filtering**. Approve this final prompt to proceed.
```

> - `11.0`, `12.0`: Users cannot proceed with installation without first granting the sensor the ability to load a <<system-extension, system extension>>. During installation, you will be prompted to go to **System Preferences** and approve loading the system extension. Upon approval, a second prompt appears to enable **Network Filtering**. Approve this final prompt for installation to proceed.

Reviewer: ```suggestion
- `11.0`, `12.0`: Users cannot install the Endgame sensor without allowing it to load a <<system-extension, system extension>>. During installation, you will be prompted to go to **System Preferences** and approve it. Upon approval, a second prompt appears to enable **Network Filtering**. Approve this final prompt for installation to proceed.
```

+

> You also must grant Full Disk Access to `com.endgame.systemextension`.

Reviewer 3: ```suggestion
You must also grant Full Disk Access to `com.endgame.systemextension`.
```

> You also must grant Full Disk Access to `com.endgame.systemextension`.

Reviewer: ```suggestion
You must also grant Full Disk Access to `com.endgame.systemextension`.
```

- `10.14.6+, 10.15, 11.0, 12.0`: Grant the <<endpoint-endgame-sensor, esensor file>> Full Disk Access.

NOTE: The following instructions apply to the Endgame sensor only. To see Full Disk Access requirements for the {elastic-endpoint}, refer to <<enable-fda-endpoint>>.

[discrete]
[[kernel-ext-approval-endgame]]
== Approve the kernel extension for the Endgame sensor

> For endpoints running macOS Mojave (10.14) and earlier, {elastic-endpoint} will attempt to load a kernel extension (as opposed to a system extension) during installation. This kernel extension is required to provide insight into system events such as process events, file system events, and network events. The following prompt appears during installation:

Reviewer 4: We need to be very explicit that this ONLY applies to 7.15 and below. Since this is docs for 8.0, should we just drop this section entirely @ferullo?

> For endpoints running macOS Mojave (10.14) and earlier, {elastic-endpoint} will attempt to load a kernel extension (as opposed to a system extension) during installation. This kernel extension is required to provide insight into system events such as process events, file system events, and network events. The following prompt appears during installation:

Reviewer: I could be missing something but I'm surprised to see {elastic-endpoint} since it said above this is only for the Endgame sensor

--

> image::images/fda/sys-ext-blocked.png[System extension blocked]

Reviewer 2: Nitpick, but this image has an errant mouse cursor in it

--

To approve the extension:

. Click *Open Security Preferences*.
. In the lower-left corner of the pane, click the **Lock button**, then enter your credentials to authenticate.
. Click *Allow* to load the kernel extension.
+
--
image::images/fda/allow-kernel-ext.png[Allow kernel extension]
--
+

If the prompt does not appear, enable the extension by doing the following:

. Open a Terminal application.
. Enter `kextload /Library/Extension/kendpoint.kext`. Prepend the command with `sudo` if necessary.

> You should receive an output similar to `149    0 0xffffff7f82e7b000 0x21000    0x21000    co.elastic.kendpoint (7.11.0) BD152A57-ABD3-370A-BBE8-D15A0FCBD19A <6 5 2 1>`. If you receive this output, the kernel extension is enabled.

Reviewer: ```suggestion
You should receive output similar to `149    0 0xffffff7f82e7b000 0x21000    0x21000    co.elastic.kendpoint (7.11.0) BD152A57-ABD3-370A-BBE8-D15A0FCBD19A <6 5 2 1>`. If you receive this output, the kernel extension is enabled.
```

[discrete]
[[system-extension]]
== Approve the system extension for the Endgame sensor

> (I need screenshots for all these steps)

Reviewer 2: Remove note to self?
